### DISCUSSION TITLE
JAVA api

### DISCUSSION THREAD

**Discussion by ssszzzlll** [alexklibisz/elastiknn#395] [Q&A] [answered]
### Background

"elastiknn_nearest_neighbors" is a query type, but I can't find it in JAVA API, Method in QueryBuilders does not include it

### Deliverables

GET /my-index/_search
{
    "query": {
        "elastiknn_nearest_neighbors": {       
            "field": "my_vec",                 
            "vec": {                           
                "values": [0.1, 0.2, 0.3, ...],               
            },
            "model": "exact",                  
            "similarity": "cosine",            
            ...                                 
        }
    }
}

**Comment by alexklibisz** [comment, marked as answer]
There's was a Java client library, with a query builder, but I removed it in this PR: https://github.com/alexklibisz/elastiknn/pull/399 . For more context, see #392.

You should still be able to use the code as an example, or just copy paste it, to create your own Java client.
